### PR TITLE
[BYTEMAN-415] removing recursive structure for thread history helper MXBean to work in JDK11

### PR DESCRIPTION
**Pull request by ochaloup**
plus enhancing the structure with Callable handling

https://issues.redhat.com/browse/BYTEMAN-415

### CODE REVIEW

**Comment by ochaloup** [comment]
For MXBean could be registered I'm removing the recursive structure from `ThreadMonitored` object and saving only the information who created the thread as a `String`. The information is still possible to be found as the String contains the Id and all events can be listed and filtered for the one needed.

File: sample/src/org/jboss/byteman/sample/helper/ThreadHistoryMonitorHelper.java [hunk 1]
```diff
@@ -101,8 +103,9 @@ public void registerHelperMBean(String name) {
             MBeanServer mbs = ManagementFactory.getPlatformMBeanServer();
             try {
                 ObjectName oname = new ObjectName(name);
-                if(mbs.isRegistered(oname) == false)
+                if(mbs.isRegistered(oname) == false) {
```
Comment (adinn) [inline sample/src/org/jboss/byteman/sample/helper/ThreadHistoryMonitorHelper.java:106]: space needed after if

**Review by adinn — APPROVED** [review]
This looks fine, Ondra. Thanks for updating it.

File: sample/src/org/jboss/byteman/sample/helper/ThreadHistoryMonitorHelper.java (hunk 1 above)
Comment (ochaloup) [reply sample/src/org/jboss/byteman/sample/helper/ThreadHistoryMonitorHelper.java]: Updated.

**Pull request merged**